[IMP] accounting: Deferred revenues and expenses (#6455)
odoo/documentation: content/applications/finance/accounting/customer_invoices/deferred_revenues.rst

These future revenues must be deferred on the company's balance sheet among the current liabilities
until they can be **recognized**, at once or over a defined period, on the Profit and Loss statement.

For example, let's say we sell a five-year extended warranty for $ 350. We already receive the money
now but haven't earned it yet. Therefore, we post this new income in a deferred revenue account and
decide to recognize it on a yearly basis. Each year, for the next 5 years, $ 70 will be recognized
as revenue.
For example, let's say we sell a software license of $ 1200 for 1 year.
We immediately invoice it to the customer but can't considered having it earned it yet.
Therefore, we post this new revenue in a deferred revenue account and
decide to recognize it on a monthly basis. Each month, for the next 12 months, $ 100 will be
recognized as an revenue.

Odoo Accounting handles deferred revenues by spreading them in multiple entries that are
automatically created in *draft mode* and then posted periodically.
posted periodically.

.. note::
   The server checks once a day if an entry must be posted. It might then take up to 24 hours before

>    you see a change from *draft* to *posted*.

Reviewer: This is still true depending on the settings, sure we want to remove? (if not, I think that "Odoo" is better than "The server")

Author: We decided to remove it with Benj. This is an explanation of the auto-post mechanism not deferred

    Revenues will be deferred on this Current Liability account until they are recognized.
- Generate Entries
    By default, Odoo will :ref:`automatically generate <customer_invoices/deferred/generate_on_validation>` the deferral entries when
    you post a customer invoice. However,
    you can also choose to :ref:`generate them manually <customer_invoices/deferred/generate_manually>`
    by selecting the :guilabel:`Manually & Grouped` option instead.
- Amount Computation
    Suppose you have an invoice of $ 1200 that must
    be deferred over 12 months, the :guilabel:`Equal per month` computation will account for $ 100 each month. While
    the :guilabel:`Based on days` computation will account amounts depending on the number
    of days in each month.

.. _customer_invoices/deferred/generate_on_validation:

Generate deferral entries on validation
=======================================

Make sure the :guilabel:`Start Date` and :guilabel:`End Date` fields are visible as shown below:

.. image:: deferred_revenues/deferred_revenue_show_start_end_dates_fields.png
   :align: center
   :alt: Show the Start and End Date fields

> For each line of the invoice, dated in January, you can then specify the start and

Reviewer 2: Makes sense to me :) Thanks for the explanation

> For each line of the invoice, dated in January, you can then specify the start and

Reviewer 2: @RicGR98 I accidentally left this one as resolved even though I had a question/suggestion for it. What do you think?

> For each line of the invoice, dated in January, you can then specify the start and

Author: Sounds good to me ! It's easier to read indeed :)

From March to October
~~~~~~~~~~~~~~~~~~~~~

The same computation is done for each month until October.

In November
~~~~~~~~~~~

At the end of November, after clicking the :guilabel:`Generate Entries` button, we will have the following entries:

- Entry 1 dated on the 30th November:

  - Line 1: Expense account -1200 -600 = **-1800** (we cancel the total of both invoices)
  - Line 2: Expense account 1100 + 550 = **1650** (we recognize 11/12 of invoice A and 11/12 of invoice B)
  - Line 3: Deferred account 1800 - 1650 = **150** (amount that has yet to be deferred later on)

- Entry 2 dated on the 1st December, the reversal of the previous entry.

In December
~~~~~~~~~~~

There is no need to generate entries in December. Indeed, if we do the computation for December,
we will have an amount of 0 to be deferred.

In total
~~~~~~~~

If we aggregate everything, we would have:

- invoice A and invoice B
- 2 entries (one for the deferral and one for the reversal) for each month from January to November

Therefore, at the end of December, invoices A and B are fully recognized as income,
only once thanks to the reversal mechanism.

> Why would you need the "Manually & Grouped" mode?
> -------------------------------------------------
> 

Reviewer 2: This section is very useful, but we usually avoid FAQ sectionsin the documentation. Instad let's put this in a tip admonition block (.. tip::) and remove this line:
```suggestion
.. tip::
```

> Suppose you have **1000** invoices.

Reviewer 2: ```suggestion
  Suppose a business has **1000** invoices.
```
Remove the second person pronouns ("you"/"your") in this section.

> In the *On validation* mode, you would have:
> 1000 invoices × 13 deferral entries (reversal of the original invoices + 12 months) = **13000** deferral entries

Reviewer 2: ```suggestion
  In the *On invoice/bill validation* mode, there would be:

  1000 invoices × 13 deferral entries (reversal of the original invoices + 12 months) = **13000** deferral entries
```

> In the *Manually & grouped* mode, we do not depend on the number of entries
> as they are all aggregated into one, but only on the periods, so you would have:
> 2 entries (deferral + reversal) × 11 months = **22** deferral entries

Reviewer 2: ```suggestion
  In the *Manually & Grouped* mode, the number of deferred revenues does not affect the number of
  journal entries as they are all aggregated into one entry. Instead, the number of journal entries
  is determined by the number of periods, so there would be:

  2 entries (deferral + reversal) x 11 months = **22** deferral entries
```

> Therefore *Manually & Grouped* mode might be useful when you have lots of deferred revenues
> and you want to reduce the number of journal items created by the deferrals.

Reviewer 2: ```suggestion
  Therefore *Manually & Grouped* mode is useful when there are lots of deferred revenues and you
  want to reduce the number of journal items created by the deferrals.
```
In my opinion, leaving this "you" is okay because we're talking about if this is the user's specific desire instead of talking about a general example.

   and an end date of 12/31/2023. At the end of August, $800 is recognized as an income,
   whereas $400 remains on the deferred account.

Reporting
=========

The deferred revenue report computes an overview of the necessary deferral entries for each account.
To access it, go to :menuselection:`Accounting --> Reporting --> Deferred Revenue`.

To view the journal items of each account, click on the account name and then :guilabel:`Journal
Items`.

.. image:: deferred_revenues/deferred_revenue_report.png
   :alt: Deferred revenue report

.. note::
    Only invoices whose accounting date is before the end of the period of the report
    are taken into account.

.. _customer_invoices/deferred/generate_manually:

Generate grouped deferral entries manually
==========================================

If you have a lot of deferred revenues and wish to reduce the number of journal entries created, you
can generate deferral entries manually. To do so, set the :guilabel:`Generate Entries` field in the
**Settings** to :guilabel:`Manually & Grouped`. Odoo then aggregates the deferred amounts in a

> single entry.

Reviewer 3: I used the contents of the tip that you had at the end of the file here instead.  I felt is was a) not really a tip and b) more useful here. I also removed the "example" part of the tip as I felt it wasn't necessary.

Deferred revenues
=================

**Deferred revenues**, or **unearned revenues**, are invoices addressed to customers
for goods yet to be delivered or services yet to be rendered.

The company cannot report them on the current **profit and loss statement**, or *income statement*,
since the goods and services will be effectively delivered/rendered in the future.

These future revenues must be deferred on the company's balance sheet among the current liabilities
until they can be **recognized**, at once or over a defined period, on the profit and loss statement.

For example, let's say a business sells a software license of $1200 for 1 year. They immediately
invoice it to the customer but can't consider it earned yet, as the future months of licensing have
not yet been delivered. Therefore, they post this new revenue in a deferred revenue account and
recognize it on a monthly basis. Each month, for the next 12 months, $100 will be recognized as
revenue.

Odoo Accounting handles deferred revenues by spreading them in multiple entries that are
posted periodically.

.. note::
   The server checks once a day if an entry must be posted. It might then take up to 24 hours before

>    you see a change from :guilabel:`draft` to :guilabel:`posted`.

Reviewer 2: ```suggestion
   you see a change from :guilabel:`Draft` to :guilabel:`Posted`.
```
